File: Project_A/src/CharCount.java
// charcount.c - Count characters in a sourcefile

import java.io.File;
import java.io.FileInputStream;
import java.io.IOException;

class CharCount {

    private static int EOF = -1;
    private static File srcFile = null;
    private static String srcFilename = "<srcFilename>";

    public static void main(String[] args) throws IOException {

        if (args.length != 1) {
            System.out.println("Usage: charcount <src>\n");
            return;
        }

        if (args[0] != null) { // Check <src>
            srcFilename = args[0];
            System.out.println("copy: srcFilename = '" + srcFilename + "'");
            srcFile = new File(srcFilename);
            if (!srcFile.canRead()) {
                System.out.println("charcount: Cannot open srcFile '" + srcFilename + "'");
                return;
            }
        } else {
            System.out.println("charcount: [OK] srcFilename = '" + srcFilename + "'");
        }

        final FileInputStream srcStream = new FileInputStream(srcFile);

        // Execute the character count.
        int c;
        int nChars = 0;

        while ((c = srcStream.read()) != EOF) {
            nChars++;
        }

        srcStream.close();

        System.out.printf("%d characters\n", nChars);

    }

}

File: Project_A_V2/srcFilesForStudents_ProjectA/charcount.c
// charcount.c - Count characters in a sourcefile

#include <stdio.h>

int main(int argc, char* args[]) {
    FILE* srcFile = NULL;
    char* srcFilename = "<srcFilename>";

    // Check the command-line.
    if (argc != 2) {
        printf("Usage: charcount <src>\n");
        return 0;
    }

    // Check if arguments are valid, if the srcFile exists, and if can create the dstFile.
    if (args[1] != NULL) { // Check <src>
        srcFilename = args[1];
//      printf("charcount: srcFilename = '%s'\n", srcFilename);
        srcFile = fopen(srcFilename, "r");
        if (srcFile == NULL) {
            printf("charcount: Cannot open srcFile '%s'\n", srcFilename);
            return 0;
        }
    } else {
        printf("charcount: [OK] srcFilename = '%s'\n", srcFilename);
    }

    // Execute the character count.
    int c;
    int nChars = 0;

    while ((c = fgetc(srcFile)) != EOF) {
        ++nChars;
    }

    // Close and flush the file.
    fclose(srcFile);

    // Print the result.
    printf("%d characters\n", nChars);
    return 1;
}
